R/watcher.R

```diff
@@ -2,53 +2,56 @@
 #'
 #' @param debug activate debug mode where output will be both printed to
 #'   screen and captured.
+#' @param handler An ouptut handler object.
+#' @param frame When this frame terminates, the watcher will automatically close.` 
 #' @return list containing four functions: `get_new`, `pause`,
 #'  `unpause`, `close`.
 #' @keywords internal
-watchout <- function(debug = FALSE) {
-  output <- character()
-  prev   <- character()
-
-  con <- textConnection("output", "wr", local = TRUE)
+watchout <- function(handler = new_output_handler(),
+                     debug = FALSE,
+                     frame = parent.frame()) {
+  con <- file("", "w+b")
```

Comment:
A `file()` connection takes a bit longer to set up than a text connection, but it is faster to append too (especially as the output grows long since appending to a character vector is `O(n^2)`). It's also simpler and faster to read from, and any set up time is now amortised over all top level expressions.